lib/extensions/permissions.js

```
@@ -82,22 +94,84 @@ async function execWix (args) {
  * @param {string} bundleId - bundle identifier of the target application.
- * @param {Object} permissionsMapping - An object, where keys ar  service names
- * and values are corresponding state values. See https://github.com/wix/AppleSimulatorUtils
+ * @param {Object} permissionsMapping - An object, where keys are service names
+ * and values are corresponding state values. Services listed in APPLE_PRIVACY_CMD
+ * will be set with `xcrun simctl privacy` command by Apple otherwise AppleSimulatorUtils by WIX.
+ * See the result of `xcrun simctl privacy` and https://github.com/wix/AppleSimulatorUtils
  * for more details on available service names and statuses.
+ * Note that the `xcrun simctl privacy` command kill the app process.
  * @throws {Error} If there was an error while changing permissions.
  */
-async function setAccess (udid, bundleId, permissionsMapping) {
-  const permissionsArg = _.toPairs(permissionsMapping)
-    .map((x) => `${x[0]}=${formatStatus(x[1])}`)
-    .join(',');
-  return await execWix([
-    '--byId', udid,
-    '--bundle', bundleId,
-    '--setPermissions', permissionsArg,
-  ]);
+async function setAccess (simctl, udid, bundleId, permissionsMapping) {
+  const wixPermissions = {};
+
+  const grantPermissions = [];
+  const revokePermissions = [];
+  const resetPermissions = [];
+
+  for (const serviceName in permissionsMapping) {
+    if (!APPLE_PRIVACY_CMD.includes(serviceName)) {
+      wixPermissions[serviceName] = permissionsMapping[serviceName];
+    } else {
+      switch (permissionsMapping[serviceName]) {
+        case STATUS.YES:
+          grantPermissions.push(serviceName);
+          break;
+        case STATUS.NO:
+          revokePermissions.push(serviceName);
+          break;
+        case STATUS.UNSET:
+          resetPermissions.push(serviceName);
+          break;
+        default:
+          log.warn(`${serviceName} does not support ${permissionsMapping[serviceName]}. Please specify yes, no or unset.`);
```

I assume we should throw an exception here